lib/src/clojure_lsp/queries.clj

```diff
@@ -316,45 +316,32 @@
           (get analysis filename))))
 
 (defmethod find-references :var-usages
-  [analysis element include-declaration? _db]
+  [analysis element include-declaration? db]
   (if (= (:to element) :clj-kondo/unknown-namespace)
     [element]
-    (into []
-          (comp
-            (mapcat val)
-            (filter #(not (identical? :keywords (:bucket %))))
-            (filter #(safe-equal? (:name element) (:name %)))
-            (filter #(safe-equal? (:to element) (or (:ns %) (:to %))))
-            (filter #(or include-declaration?
-                         (and (or (not (:from-var %))
-                                  (not= (:from-var %) (:name element))
-                                  (not= (:from %) (:to element)))
-                              (not (identical? :var-definitions (:bucket %)))
-                              (not (:defmethod %)))))
-            (medley/distinct-by (juxt :filename :name :row :col)))
-          analysis)))
+    (let [var-definition {:ns (:to element)
+                          :name (:name element)
+                          :bucket :var-definitions}]
+      (find-references analysis var-definition include-declaration? db))))
 
 (defmethod find-references :var-definitions
   [analysis element include-declaration? _db]
-  (let [defrecord? (= 'clojure.core/defrecord (:defined-by element))
-        deftype? (= 'clojure.core/deftype (:defined-by element))
-        names (cond
-                defrecord? (defrecord-names-for element)
-                deftype? (deftype-names-for element))]
+  (let [names (var-definition-names element)
+        exclude-declaration? (not include-declaration?)]
     (into []
           (comp
             (mapcat val)
-            (filter #(not (identical? :keywords (:bucket %))))
-            (filter #(or (safe-equal? (:name element) (:name %))
-                         (and (or defrecord? deftype?)
-                              (contains? names (:name %)))))
+            (remove #(identical? :keywords (:bucket %)))
+            (filter #(contains? names (:name %)))
             (filter #(safe-equal? (:ns element) (or (:ns %) (:to %))))
-            (filter #(or include-declaration?
-                         (and (or (not (identical? :var-definitions (:bucket %)))
-                                  (= (:defined-by %) 'clojure.core/declare))
-                              (or (not (:from-var %))
-                                  (not= (:from-var %) (:name element))
-                                  (not= (:from %) (:ns element))))))
+            (remove #(and exclude-declaration?
```

Comment:
Shouldn't this be a `or`?
```suggestion
            (remove #(or exclude-declaration?
```

Reply:
I don't think so it would be clearer, sounds the same to me 🤷🏻‍♂️  but I'm ok too, I think this was `exclude-declaration` before I try to make it clearer with `include-declaration` but we can see no matter how, it's hard indeed 😂